docs/calendar-draft.md

```diff
@@ -0,0 +1,315 @@
+
+	[Temporal.Calendar.id] = "iso",
+
+	// ALL OTHER METHODS:
+	[Temporal.Calendar.plus] = () => {
+		throw new TypeError("Unsupported operation: full calendar required");
+	}
+	// Same for [Temporal.Calendar.minus], etc.
+}
+```
+
+It would in effect render default Temporal.Date (and Temporal.DateTime) with fewer operations until you specify a calendar.  The following methods/getters would throw:
+
+- .dayOfWeek
+- .weekOfYear
+- .daysInMonth
+- .daysInYear
+- .leapYear
+- .plus() -- *might* be OK if the programmer requests only time units
+- .minus() -- *might* be OK if the programmer requests only time units
+- .difference() -- *might* be OK if the programmer requests only time units
+- .getYearMonth()
+- .getMonthDay()
+
+The following methods/getters would still work:
+
+- .with()
+- .withTime()
+- .toString()
+- .toLocaleString()
+- .compare()
+
+Although small, this set of operations still covers many of the recipes in the Temporal Cookbook.
+
+To enable the extended set of operations, the user would just use `.withCalendar()`:
+
+```javascript
+// Force the Gregorian calendar:
+Temporal.Date.from("2019-12-06").withCalendar("gregory").weekOfYear;
+
+// Use a calendar from another source:
+Temporal.Date.from("2019-12-06").withCalendar(Intl.defaultCalendar).weekOfYear;
+Temporal.Date.from("2019-12-06").withCalendar(request.calendar).weekOfYear;
+```
+
+The Gregorian calendar, ID "gregory" ([why not "gregorian"?](https://github.com/tc39/ecma402/issues/212)), could be used to indicate the full ISO calendar.  So, "iso" would mean the partial version, and "gregory" would mean the full version.  Alternatively, "iso8601", the identifier defined by CLDR as "Gregorian calendar using the ISO 8601 calendar week rules", could be the identifier for the full ISO calendar.
+
+## Temporal.Date API changes
+
+### New Temporal.Date instance methods
+
+The following methods involving ISO conversion would be added:
+
+```javascript
+Temporal.Date.prototype.toISO = function(): Temporal.Date {
+	const isoDate = this.calendar[Temporal.Calendar.toISO](this);
+	// assert: isoDate.calendar === Temporal.Calendar.iso
+	return isoDate;
+}
+
+// Temporal.Date.prototype.with does *not* modify the calendar. A new method
+// is added for that:
+
+Temporal.Date.prototype.withCalendar = function(newCalendar: Calendar) {
+	const isoDate = this.toISO();  // note: call intrinsic version
+	const otherDate = newCalendar[Temporal.Calendar.fromISO](isoDate);
+	// assert: otherDate.calendar === newCalendar
+	return otherDate;
+}
+```
+
+### ISO strings with calendar hint
+
+Serialization to/from strings will still be supported.  To convert to a string, the `Temporal.Calendar.id` field of the calendar will be appended in `[]` following the date.  For example, `2019-12-06[hebrew]` refers to 2019-12-06 projected into the Hebrew calendar.
+
+```javascript
+Temporal.Date.prototype.toString = function() {
+	let calendarKeyword, isoDate;
+	if (this.calendar === Temporal.Calendar.iso) {
```

Comment:
This kind of special casing is hazardous. What if we have an ISO 8601 calendar from a different realm, or one that is different from the default only by installation of spy operations?

Reply:
Cross-realm would be achieved either with an internal slot (my preference) or with a well-known symbol.

Reply:
This should never happen: it would always be

if (other.calendar !== this.calendar) {
other = this.calendar.fromISO(other.calendar.toISO(other));
}

Reply:
I’d hope that calendar objects aren’t just glorified serialized strings; I’d expect internal slot access here.

Reply:
They are key objects that codify the calendar rules and enable translations and arithmethic.

Reply:
They are key objects that codify the calendar rules and enable translations and arithmethic.

Reply:
I fixed the if statement to replace the `===` with prose to fix @gibson042's comment.

The identity/protocol debate is going on in the other thread.  Resolving this thread.